cmd/gossamer/account.go

```diff
@@ -0,0 +1,261 @@
+			return err
+		}
+	}
+
+	// check if we want to generate a new keypair
+	// can specify key type using --ed25519 or --sr25519
+	// otherwise defaults to sr25519
+	if keygen := ctx.Bool(utils.GenerateFlag.Name); keygen {
+		log.Info("generating keypair...")
+
+		// check if --ed25519 or --sr25519 is set
+		keytype := utils.Sr25519KeyType
+		if flagtype := ctx.String(utils.Sr25519Flag.Name); flagtype != "" {
+			keytype = flagtype
+		} else if flagtype := ctx.String(utils.Ed25519Flag.Name); flagtype != "" {
+			keytype = flagtype
+		}
+
+		// check if --password is set
+		var password []byte = nil
+		if pwdflag := ctx.String(utils.PasswordFlag.Name); pwdflag != "" {
+			password = []byte(pwdflag)
+		}
+
+		_, err = generateKeypair(keytype, datadir, password)
+		if err != nil {
+			log.Error("generate error", "error", err)
+			return err
+		}
+	}
+
+	// import key
+	if keyimport := ctx.String(utils.ImportFlag.Name); keyimport != "" {
+		log.Info("importing key...")
+		_, err = importKey(keyimport, datadir)
+		if err != nil {
+			log.Error("import error", "error", err)
+			return err
+		}
+	}
+
+	// list keys
+	if keylist := ctx.Bool(utils.ListFlag.Name); keylist {
+		_, err = listKeys(datadir)
+		if err != nil {
+			log.Error("list error", "error", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+// importKey imports a key specified by its filename to datadir/keystore/
+// it saves it under the filename "[publickey].key"
+// it returns the absolute path of the imported key file
+func importKey(filename, datadir string) (string, error) {
+	keystorepath, err := keystoreDir(datadir)
+	if err != nil {
+		return "", fmt.Errorf("could not get keystore directory: %s", err)
+	}
+
+	importdata, err := ioutil.ReadFile(filepath.Clean(filename))
+	if err != nil {
+		return "", fmt.Errorf("could not read import file: %s", err)
+	}
+
+	ksjson := new(keystore.EncryptedKeystore)
+	err = json.Unmarshal(importdata, ksjson)
+	if err != nil {
+		return "", fmt.Errorf("could not read file contents: %s", err)
+	}
+
+	keystorefile, err := filepath.Abs(keystorepath + "/" + ksjson.PublicKey[2:] + ".key")
+	if err != nil {
+		return "", fmt.Errorf("could not create keystore file path: %s", err)
+	}
+
+	err = ioutil.WriteFile(keystorefile, importdata, 0644)
+	if err != nil {
+		return "", fmt.Errorf("could not write to keystore directory: %s", err)
+	}
+
+	log.Info("successfully imported key", "public key", ksjson.PublicKey, "file", keystorefile)
+	return keystorefile, nil
+}
+
+// listKeys lists all the keys in the datadir/keystore/ directory and returns them as a list of filepaths
+func listKeys(datadir string) ([]string, error) {
+	keystorepath, err := keystoreDir(datadir)
+	if err != nil {
+		return nil, fmt.Errorf("could not get keystore directory: %s", err)
+	}
+
+	files, err := ioutil.ReadDir(keystorepath)
+	if err != nil {
+		return nil, fmt.Errorf("could not read keystore dir: %s", err)
+	}
+
+	keys := []string{}
+
+	for _, f := range files {
```

Comment:
This prints all files not just keys

Reply:
okay, I have changed it to check for the .key extension